### ISSUE TITLE
`google-chrome` rendering broken on wayland, version M124 (124.0.6367.60)

### ISSUE THREAD

**Issue by JohnRTitor** [NixOS/nixpkgs#306010]
### Describe the bug
Instead of a normal Chrome interface, a blank window appears

### Steps To Reproduce
1. Install `google-chrome` 124.0.6367.60 from unstable 
2. Launch `google-chrome-stable`
3. Instead of seeing a normal Chrome window you see a blank window

### Expected behavior
A normal Chrome new tab page.

### Screenshots
![image](https://github.com/NixOS/nixpkgs/assets/50095635/6f8db165-3bf6-49c1-8bda-ea5d5eaed712)

Note: the transparency and blur is my Window rule of my tiling window manager Hyprland. Chrome is still unusable even after disabling that.

### Additional context
I suspect this is related to recent [Mesa update](https://github.com/NixOS/nixpkgs/pull/301941).

This happened after #305145 got merged and mesa 24.0.5 hit unstable yesterday.
I suspect this is either of those problem.

### Notify maintainers

@emilylange @networkException 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"

 - system: `"x86_64-linux"`
 - host os: `Linux 6.8.7-cachyos, NixOS, 24.05 (Uakari), 24.05.20240421.6143fc5`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/cc0ribc27c349w1v2anrby7z8kigy5pq-source`

```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by JohnRTitor** [comment]
Thanks for the reply. I wasn't sure who to ping since I couldn't find a specific PR of Chrome update, then I realised Chrome inherits, [Chromium platform version](https://github.com/NixOS/nixpkgs/blob/6143fc5eeb9c4f00163267708e26191d1e918932/pkgs/applications/networking/browsers/google-chrome/default.nix#L55) and Chrome is updated along with Chromium PRs.

And the maintainer of `google-chrome` apparently is @primeos.

As a temporary workaround, I [pinned my google-chrome](https://github.com/JohnRTitor/nix-conf/commit/b8577c22302dfd1226c1e55a634a10c8ef110e6e) to https://github.com/NixOS/nixpkgs/commit/2e359fb3162c85095409071d131e08252d91a14f, which is the last revision of nixpkgs to have Chrome 123.

Please do update the Chromium package as soon as the fix lands in a release, since `google-chrome` package inherits its version from Chromium.

I'll let this issue stay open as a tracker, and for those searching for any workaround.

**Comment by networkException** [comment]
> Please do update the Chromium package as soon as the fix lands in a release, since google-chrome package inherits its version from Chromium.

As I said in my previous comment we already include a fix for this in the `chromium` package: https://github.com/NixOS/nixpkgs/commit/fb10ea41b2a080cb4ecac34aa12cf056d8d3f90e

**Comment by JohnRTitor** [comment]
I was talking about incrementing base Chromium version.

**Comment by alyraffauf** [comment]
Having the same issue. As a workaround, if you launch Chrome with `--ozone-platform=wayland` (assuming you are on wayland), it works just fine. Somehow, this is distinct from manually setting the ozone flag to wayland, but it's working in the meantime while we wait for an update

**Comment by JohnRTitor** [comment]
@networkException @emilylange is this fixed on `124.0.6367.78`?

**Comment by networkException** [comment]
No, Google started including the patch for M126 versions.

**Comment by emilylange** [comment]
https://github.com/NixOS/nixpkgs/issues/306471#issuecomment-2075022049 might be worth a read for those of you that may consider stepping up to maintain `google-chrome` :)

I would be heavily in favor of making `google-chrome` independent of `chromium`.

We (@networkException and I, the current and only `chromium` maintainers) are already at capacity with `chromium` (and `ungoogled-chromium`) maintenance.
We have by no means time to maintain `google-chrome` as well.

FYI: `google-chrome` is a binary package, unlike `chromium`, which is built from source.
Meaning `google-chrome` is extremely low maintenance, as can be seen from the commit log.

It just needs version bumps.
In theory, splitting the packages might also allow for quicker `google-chrome` bumps, since `chromium` may take a bit longer from time to time to fix some issues with our source build before it can be merged.

**Comment by JohnRTitor** [comment]
I think I'll throw my hat at this one :)

**Comment by fiq** [comment]
Is there a wayu to leave this pkg alone pinned to the last version of the google-chrome binary, until such a time as Google push an upstream fix?

**Comment by JohnRTitor** [comment]
The ideal scenario would be google-chrome independent of chromium versions.

It would allow detecting regressions like this, which obviously can't be solved with a patch (this is a prebuilt binary). The release can be pinned to the unaffected version and not be updated until there is a upstream release with the fix.

As a side effect of #305145, the affected google-chrome release have already reached the "stable-23.11" release.

@emilylange IMHO, it should be decoupled before the 24.05 release, else we may see similar situations in the foreseeable future.

Note: this bug has not yet been fixed on stable `124.0.6367.91`. The fix appears to have been landed in beta `125.0.6422.14`.

**Comment by selfisekai** [comment]
it got merged into the m124 branch so it should be part of release tomorrow https://chromium-review.googlesource.com/c/chromium/src/+/5497377

**Comment by selfisekai** [comment]
here it is https://chromereleases.googleblog.com/2024/04/stable-channel-update-for-desktop_30.html

**Comment by emilylange** [comment]
I find it a bit sad that nobody has come forth so far to maintain `google-chrome`.

I can help to untie `google-chrome` from the `chromium` derivation, to kick-start things, if that's the issue.

Honestly, I might do that either way, considering that if worse comes to worst, @r-ryantm (https://github.com/nix-community/nixpkgs-update) will simply auto-bump `google-chrome`.

And if not even @r-ryantm ends up bumping `google-chrome`, I'll have to resort to marking `google-chrome` as insecure using `meta.knownVulnerabilities`.

With all of that out of the way:
There is #308358 now, which bumps `chromium` and by proxy `google-chrome` to `124.0.6367.118`.

After this PR has been merged, it can be tracked using https://nixpk.gs/pr-tracker.html?pr=308358 (or similar).
It will also get backported to `23.11`.

**Comment by jnsgruk** [comment]
@emilylange i can try to help with the version bumps etc.

Chrome is a lot bigger and more complicated than anything I've taken on before, but I'm happy to help take care of the "boring stuff" and take advice on the harder bits where needed.

I think splitting it would be a good starting point for sure.

**Comment by emilylange** [comment]
Thank you :)

I will start working on separating `google-chrome` and `chromium` and will ping you when I am done.

**Comment by massix** [comment]
Hello, just FIY, the same issue exists with Microsoft Edge (derivation `microsoft-edge-stable`) since they're using the same engine as Chrome

**Comment by emilylange** [comment]
#308358 and #308401 landed in their according channels by now.

#308383 and #308793 are essentially a no-op, so no need to fair for those two.

Please report future issues with `google-chrome` to the new `google-chrome` maintainer @jnsgruk.

Closing.

---
PS:

> Hello, just FIY, the same issue exists with Microsoft Edge (derivation `microsoft-edge-stable`) since they're using the same engine as Chrome

Please open a new issue and ping the `microsoft-edge-stable` maintainers.
